lib/improver/tests/metadata/test_utilities.py

```diff
@@ -63,26 +67,29 @@ def test_dictionary_input(self):
 
         hash_input = {'one': 1, 'two': 2}
         result = generate_hash(hash_input)
-        expected = "4735f4a74dd17d27b383de504a87e324"
+        expected = (
+            "c261139b6339f880f4f75a3bf5a08f7c2d6f208e2720760f362e4464735e3845"
+        )
         self.assertIsInstance(result, str)
         self.assertEqual(result, expected)
 
-    def test_dictionary_order_variant(self):
-        """Test the expected hash is different if the dictionary order is
-        different."""
+    def test_dictionary_order_invariant(self):
+        """Test the expected hash is the same for different dict ordering."""
 
         hash_input1 = {'one': 1, 'two': 2}
         hash_input2 = {'two': 2, 'one': 1}
         result1 = generate_hash(hash_input1)
         result2 = generate_hash(hash_input2)
-        self.assertNotEqual(result1, result2)
+        self.assertEqual(result1, result2)
```

Comment:
This behaviour change is due to the `sort_dicts=True` default in pprint, which I think is sensible, since metadata comparison shouldn't be order dependent. It may be worth explicitly setting `sort_dicts` in the `pprint.pformat()` call rather than using the implicit default to make it more obvious that is the intention.

Reply:
Thanks, now done